From b4d48fcdc3f73ec0d335024abd836bf13fea3959 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Sat, 9 Dec 2023 10:24:54 +0100
Subject: [PATCH 04/23] parisc/scsi: Add maximum transfer size per SCSI
 controller

LSI controller allows up to 4MB transactions per call, the ESP controller only
up to 64k per call.
Values for other drivers need to be added too.

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/block.h       | 1 +
 src/hw/esp-scsi.c | 1 +
 src/hw/lsi-scsi.c | 1 +
 3 files changed, 3 insertions(+)

diff --git a/roms/seabios-hppa/src/block.h b/roms/seabios-hppa/src/block.h
index 5cd2e5db..f780847e 100644
--- a/roms/seabios-hppa/src/block.h
+++ b/roms/seabios-hppa/src/block.h
@@ -51,6 +51,7 @@ struct drive_s {
     struct chs_s lchs;  // Logical CHS
     u64 sectors;        // Total sectors count
     u32 cntl_id;        // Unique id for a given driver type.
+    u32 max_bytes_transfer; // maximum number of bytes which can bet transferred at once
     u8 removable;       // Is media removable (currently unused)
 
     // Info for EDD calls
diff --git a/roms/seabios-hppa/src/hw/esp-scsi.c b/roms/seabios-hppa/src/hw/esp-scsi.c
index 38c95c1d..ab161de2 100644
--- a/roms/seabios-hppa/src/hw/esp-scsi.c
+++ b/roms/seabios-hppa/src/hw/esp-scsi.c
@@ -180,6 +180,7 @@ esp_scsi_init_lun(struct esp_lun_s *llun, struct pci_device *pci, u32 iobase,
 {
     memset(llun, 0, sizeof(*llun));
     llun->drive.type = DTYPE_ESP_SCSI;
+    llun->drive.max_bytes_transfer = 64*1024;   /* 64kb */
     llun->drive.cntl_id = pci->bdf;
     llun->pci = pci;
     llun->target = target;
diff --git a/roms/seabios-hppa/src/hw/lsi-scsi.c b/roms/seabios-hppa/src/hw/lsi-scsi.c
index 5583bd69..784d3e24 100644
--- a/roms/seabios-hppa/src/hw/lsi-scsi.c
+++ b/roms/seabios-hppa/src/hw/lsi-scsi.c
@@ -142,6 +142,7 @@ lsi_scsi_init_lun(struct lsi_lun_s *llun, struct pci_device *pci, u32 iobase,
 {
     memset(llun, 0, sizeof(*llun));
     llun->drive.type = DTYPE_LSI_SCSI;
+    llun->drive.max_bytes_transfer = 4*1024*1024;   /* 4 MB */
     llun->drive.cntl_id = pci->bdf;
     llun->pci = pci;
     llun->target = target;
-- 
2.39.2

